config/config.go

```diff
@@ -180,38 +181,73 @@ type Config struct {
 	OmitExtensions    bool
 }
 
+func extractMasterKeys(group keyGroup) (sops.KeyGroup, error) {
+	var mergeKeyGroup sops.KeyGroup
+	for _, k := range group.Merge {
+		subKeyGroup, err := extractMasterKeys(k)
+		if err != nil {
+			return nil, err
+		}
+		mergeKeyGroup = append(mergeKeyGroup, subKeyGroup...)
+	}
+
+	var keyGroup sops.KeyGroup
+
+	// Add only unique values from mergeGroup to keyGroup
+	unique := make(map[string]bool)
+	for _, v := range mergeKeyGroup {
+		// keytype + toMap make a unique combination per key type
+		valueMap := v.ToMap()
+		delete(valueMap, "created_at")
```

Comment:
Why not use `v.ToString()` instead of a string representation of `v.ToMap()`?

The only problem I see with that is that `kms.ToString()` only uses `arn` and not `role`, `context`, and `awsProfile`. I have no idea whether that's OK or not... From https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id it *seems* that the ARN uniquely identifies the key.

Does anyone from @getsops/maintainers have some insight here?

Reply:
This difference on (at least) the `kms` keys is the reason.

I found this other place, where keyGroups are compared to each other:
https://github.com/getsops/sops/blob/1c46d2492192cc55bc33baf6b9b78889fe3e795a/cmd/sops/common/common.go#L401
and here `ToString` is used:
https://github.com/getsops/sops/blob/1c46d2492192cc55bc33baf6b9b78889fe3e795a/cmd/sops/common/common.go#L423
To have the same behavior, I have changed my code to use `ToString` too.